### DISCUSSION TITLE
Dependencies

### DISCUSSION THREAD

**Discussion by rvanee** [Mayitzin/ahrs#17] [Ideas]
Dear mr Mayitzin,

Thank you for your excellent library. I'm trying to use it in a Kivy application, targeted towards Windows+Android+iOS. This means I need to agressively minimize dependencies. Unfortunately, your library depends on scipy and matplotlib. You wrote about your desire to remove the matplotlib dependency from your library by trimming it down to the calculations, rather than including plotting and file I/O. I would be very supportive of that move, as you can understand. 
I was wondering if it would be possible to also get rid of the scipy dependency. The current library depends on it for reading matlab .mat files in utils/io.py and for use of a sqrt function in filters/flae.py. The latter puzzles me, however, since I can't find a sqrt function in scipy, so I wonder if numpy.sqrt was meant to be used?

Do you have a target date for your next release?

Kind regards,
Rene van Ee

**Comment by Mayitzin** [comment]
Hi Renee,

Indeed, the submodules `io` and `plot` will be removed and the dependency on `matplotlib` will be totally removed.

For `scipy` it was a bit tricky, though. In the beginning I tried several datasets, and using `numpy.sqrt` in `FLAE` was giving me some trouble, because for some data it kept returning complex numbers, even after forcing non-negative values with `abs`. But `scipy.sqrt` could actually return a real number. Then, it was easier to use that function.

Now, however, I've tried again some estimations with `numpy.sqrt` and it seems to work without a problem. It returns real numbers (although, to be honest, I am using a different dataset.)

I think we can switch it back to `numpy` and get rid of scipy in that case. I'll check the rest of the package, and probably `scipy` will not be needed after all.

Just a reminder, even when the submodules `io` and `plot` are not being linked and built for the package, they will remain in the base code in the newest version (0.3) to let users know how it was being used, if they check the source. It will be entirely removed for the next release (0.4)

Thanks for pointing out the dependencies!

Mario.

**Comment by rvanee** [comment]
Hi Mario,

Thank you for your swift response. I integrated the relevant parts of the
ahrs code into my code for now, to try and get the thing working under
Kivy/Android (without scipy and matplotlib). I'll switch back to importing
your module after v0.4 is released.

Once again, thank you for your work on this!

Regards,
Rene

Op za 2 jan. 2021 om 14:28 schreef Mario Garcia <notifications@github.com>:

**Comment by rvanee** [comment]
Hi Mario,

I love version 0.3, obviously, and I use it in my kivy program for sensor
fusion. Currently, I use Tilt to get an estimate of the initial orientation
and then Madgwick. I'm currently working on sensor calibration, but the
next steps will be:
1. creation of a body reference frame (the IMU is attached to the top of
the shoe and may be rotated in every axis, and I am interested in the
orientation of the shoe sole relative to the floor).
2. sensor position estimation using linear acceleration = acceleration -
gravity.

There are a number of utilities in AHRS that have to do with frame
translation, but I'm still a bit puzzled as to what to use and how to use
it. Can you perhaps give me a few hints?

Kind regards,
Rene van Ee

Op za 2 jan. 2021 om 14:35 schreef Rene van Ee ***@***.***>:
